### ISSUE TITLE
🐛 Magic numbers with "as const" are reported as warnings

### ISSUE THREAD

**Issue by mrmattrc** [biomejs/biome#6910]
### Environment information

```block
CLI:
  Version:                      2.1.1
  Color support:                true

Platform:
  CPU Architecture:             aarch64
  OS:                           macos

Environment:
  BIOME_LOG_PATH:               unset
  BIOME_LOG_PREFIX_NAME:        unset
  BIOME_CONFIG_PATH:            unset
  BIOME_THREADS:                unset
  NO_COLOR:                     unset
  TERM:                         xterm-256color
  JS_RUNTIME_VERSION:           v24.3.0
  JS_RUNTIME_NAME:              node
  NODE_PACKAGE_MANAGER:         unset

Biome Configuration:
  Status:                       Loaded successfully
  Path:                         biome.json
  Formatter enabled:            true
  Linter enabled:               true
  Assist enabled:               true
  VCS enabled:                  true

Workspace:
  Open Documents:               0
```

### What happened?

Appending "as const" to a magic number constant still reports it as a warning. Additionally, (not sure if it should be a separate issue), but GRID_SIZE also still reports a warning with the "as const" remove due to the nested numbers.
```ts
const CELL_THICKNESS = 1 as const
const SECTION_THICKNESS = 1.2 as const
const CELL_SIZE = 50 as const
const SECTION_SIZE = 300 as const
const FADE_DISTANCE = 10_000 as const
const FADE_STRENGTH = 3.5 as const
const GRID_SIZE = [10_000, 10_000] as const
```

### Expected result

Appending "as const" to a magic number does not report a warning.

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

**Comment by siketyan** [comment]
I have no idea about if we should allow the last one or not.

**Comment by sterliakov** [comment]
The last one is at least a separate issue: `as const` should not create new diagnostics for this rule, but tuples have nothing to do with that. IMO allowing tuples has some merit, but - unlike objects - they do not have associated names, so may indeed be unclear. I'd say that defining an array/tuple of constants is a good reason to use a suppression comment, because more often than not some or all of those values can be extracted with a descriptive name.

I can fix the main part (`as const` and other type-level ops, `as number` is no worse) if no one is working on this now.
